src/clover/simulation/energy_system.py

```diff
@@ -0,0 +1,2847 @@
+                processed_total_cw_load,
+                renewable_cw_produced,
+                renewable_cw_used_directly,
+                storage_water_supplied_frame,
+                total_cw_supplied,
+                total_cw_used,
+                unmet_clean_water,
+                water_surplus_frame,
+            ]
+        )
+        if scenario.pv_t:
+            system_performance_outputs_list.extend(
+                [
+                    buffer_tank_temperature,
+                    buffer_tank_volume_supplied,
+                    clean_water_pvt_collector_output_temperature,
+                    clean_water_pvt_electric_power_per_kwh,
+                    clean_water_pvt_energy,
+                    thermal_desalination_electric_power_consumed,
+                ]
+            )
+
+    if scenario.hot_water_scenario is not None:
+        system_performance_outputs_list.extend(
+            [
+                hot_water_power_consumed,
+                hot_water_pvt_collector_output_temperature,
+                hot_water_pvt_electric_power_per_kwh,
+                hot_water_pvt_electric_power_per_unit,
+                hot_water_pvt_energy,
+                hot_water_tank_temperature,
+                hot_water_tank_volume_supplied,
+                processed_total_hw_load,
+                renewable_hw_fraction,
+            ]
+        )
+
+    system_performance_outputs = pd.concat(
+        system_performance_outputs_list,
+        axis=1,
+    )
+
+    return time_delta, system_performance_outputs, system_details
+
+
+# #%%
+# class MinigridOld:
+#     """
+#     Represents an energy system in the context of CLOVER.
+
+#     """
+
+#     def __init__(self):
+#         """
+#         Instantiate a :class:`minigrid.Minigrid` instance.
+
+#         """
+
+#         self.kerosene_data_filepath = os.path.join(
+#             self.location_filepath, "Load", "Devices in use", "kerosene_in_use.csv"
+#         )
+#         self.kerosene_usage = pd.read_csv(
+#             self.kerosene_data_filepath, index_col=0
+#         ).reset_index(drop=True)
+
+#     #%%
+#     # =============================================================================
+#     # SIMULATION FUNCTIONS
+#     #       This function simulates the energy system of a given capacity and to
+#     #       the parameters stated in the input files.
+#     # =============================================================================
+
+#     #%%
+#     # =============================================================================
+#     # GENERAL FUNCTIONS
+#     #       These functions allow users to save simulations and open previous ones,
+#     #       and resimulate the entire lifetime of a previously-optimised system
+#     #       including consideration of increasing capacity.
+#     # =============================================================================
+
+#     def lifetime_simulation(self, optimisation_report):
```

Comment:
I think this function would be useful to keep. It takes the system sizes (etc.) from the optimisation outputs and performs a simulation over the entire lifetime of the system (including accounting for the variations in component sizes), so it's quite useful for looking at the long-term system performance

Reply:
Ok, sounds good 😄 It might be good to discuss this if you've got a spare mo so we can work out how we're going to include it :)